tests/unit/mixins/should-add-token-test.js

```diff
@@ -47,3 +46,59 @@ test('does throw an exception with bad input', function (assert) {
   assert.notOk(subject.hostsMatch(undefined, 'https://server.foo.bar.com:6443/FeatureServer/0'));
   assert.notOk(subject.hostsMatch(undefined, {}));
 });
+
+test('shouldAddToken returns false when token should not be added', function (assert) {
+  const url = 'https://services2.arcgis.com/zNjnZafDYCAJAbN0/arcgis/rest/services/Street_ROW_Trees/FeatureServer/0/query?returnCountOnly=true&f=json&where=1%3D1&inSR=4326&geometry=-118.132,34.16,-118.122,34.162&geometryType=esriGeometryEnvelope&spatialRel=esriSpatialRelIntersects';
+  const serverInfo = {
+    owningSystemUrl: 'http://www.arcgis.com',
+    owningTenant: 'zNjnZafDYCAJAbN0'
+  };
+  const portalInfo = {
+    portalHostname: 'qaext.arcgis.com',
+    id: '97KLIFOSt5CxbiRI',
+    authorizedCrossOriginDomains: []
+  };
+  assert.notOk(subject.shouldAddToken(url, serverInfo, portalInfo));
+});
+
+test('shouldAddToken returns true when portal ids match', function (assert) {
+  const url = 'https://servicesqa.arcgis.com/97KLIFOSt5CxbiRI/ArcGIS/rest/services/integration_features_10001/FeatureServer/0/query?returnCountOnly=true&f=json&where=1%3D1&inSR=4326&geometry=-95.107,41.389,-95.025,41.4&geometryType=esriGeometryEnvelope&spatialRel=esriSpatialRelIntersects';
+  const serverInfo = {
+    owningSystemUrl: 'http://www.arcgis.com',
+    owningTenant: '97KLIFOSt5CxbiRI'
+  };
+  const portalInfo = {
+    portalHostname: 'qaext.arcgis.com',
+    id: '97KLIFOSt5CxbiRI',
+    authorizedCrossOriginDomains: []
+  };
+  assert.ok(subject.shouldAddToken(url, serverInfo, portalInfo));
+});
+
+test('shouldAddToken returns true when hostnames match', function (assert) {
+  const url = 'https://servicesqa.arcgis.com/97KLIFOSt5CxbiRI/ArcGIS/rest/services/integration_features_10001/FeatureServer/0/query?returnCountOnly=true&f=json&where=1%3D1&inSR=4326&geometry=-95.107,41.389,-95.025,41.4&geometryType=esriGeometryEnvelope&spatialRel=esriSpatialRelIntersects';
+  const serverInfo = {
+    owningSystemUrl: 'http://www.arcgis.com',
+    owningTenant: 'zNjnZafDYCAJAbN0'
+  };
+  const portalInfo = {
+    portalHostname: 'www.arcgis.com',
+    id: '97KLIFOSt5CxbiRI',
+    authorizedCrossOriginDomains: []
+  };
+  assert.ok(subject.shouldAddToken(url, serverInfo, portalInfo));
+});
+
+test('shouldAddToken returns true when domain is authorized', function (assert) {
+  const url = 'https://servicesqa.arcgis.com/97KLIFOSt5CxbiRI/ArcGIS/rest/services/integration_features_10001/FeatureServer/0/query?returnCountOnly=true&f=json&where=1%3D1&inSR=4326&geometry=-95.107,41.389,-95.025,41.4&geometryType=esriGeometryEnvelope&spatialRel=esriSpatialRelIntersects';
+  const serverInfo = {
+    owningSystemUrl: 'http://www.arcgis.com',
+    owningTenant: 'zNjnZafDYCAJAbN0'
+  };
+  const portalInfo = {
+    portalHostname: 'qaext.arcgis.com',
+    id: '97KLIFOSt5CxbiRI',
+    authorizedCrossOriginDomains: [ 'servicesqa.arcgis.com' ]
+  };
+  assert.ok(subject.shouldAddToken(url, serverInfo, portalInfo));
```

Comment:
this one doesn't make sense.

if the owning system is not the current portal, why are we sending a token?

Reply:
In this case the url is in `portalInfo.authorizedCrossOriginDomains`.

Reply:
I don't think that's how this works in practice. Let's say you add something to authorizedCrossOrigin, but the owning system is not yet configured or there's some weird reverse proxy. You're going to end up with an invalid token being sent

Reply:
How are you suggesting we check the `authorizedCrossOriginDomains`?

Reply:
you should only check the authorized list if the owning system matches